runtime/calamari/src/lib.rs

```diff
@@ -375,6 +376,93 @@ impl frame_system::Config for Runtime {
     type MaxConsumers = ConstU32<16>;
 }
 
+parameter_types! {
+    // One storage item; key size 32, value size 8; .
+    pub const ProxyDepositBase: Balance = deposit(1, 8);
+    // Additional storage item size of 33 bytes.
+    pub const ProxyDepositFactor: Balance = deposit(0, 33);
+    pub const AnnouncementDepositBase: Balance = deposit(1, 8);
+    pub const AnnouncementDepositFactor: Balance = deposit(0, 66);
+}
+
+/// The type used to represent the kinds of proxying allowed.
+#[derive(
+    Copy,
+    Clone,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Encode,
+    Decode,
+    RuntimeDebug,
+    MaxEncodedLen,
+    scale_info::TypeInfo,
+)]
+pub enum ProxyType {
+    Any,
+    TransferNotDie,
+    Governance,
+    Delegation,
+}
+
+impl Default for ProxyType {
+    fn default() -> Self {
+        Self::Any
+    }
+}
+
+impl InstanceFilter<RuntimeCall> for ProxyType {
+    fn filter(&self, c: &RuntimeCall) -> bool {
+        match self {
+            ProxyType::Any => true,
+            ProxyType::TransferNotDie => !matches!(
+                c,
+                RuntimeCall::Balances(pallet_balances::Call::transfer_allow_death { .. })
+            ), // proxy account cannot transfer all
```

Comment:
this will allow all calls that are not `pallet_balances::Call::transfer_allow_death { .. }`, I don't think you can use the `!matches` pattern for that reason

Reply:
unless I'm mistaken but I think we would want a sanity integration test

Reply:
`fasle` means the call will be filtered, so if it's the `transfer_allow_death`, `!matches` will be `false`. I guess it should be right.

Reply:
yeah but any other call will be `true`, this means that `TransferNotDie`  will allow you to act as a proxy for every other extrinsic except `transfer_allow_death`